packages/widget/src/hooks/useTokens.ts

```diff
@@ -48,17 +48,27 @@ export const useTokens = (selectedChainId?: number, formType?: FormType) => {
     }
 
     // Filter config tokens by chain before checking if token is allowed
-    const filteredConfigTokens = filterConfigTokensByChain(
+    const filteredConfigTokens = getConfigItemSets(
       configTokens,
-      formType,
-      selectedChainId
+      (tokens: BaseToken[]) =>
+        new Set(
+          tokens
+            .filter((t) => t.chainId === selectedChainId)
+            .map((t) => `${t.address}-${t.chainId}`)
```

Comment:
If we have already filtered tokens by chainId, why do we need to include chainId in the key? The same applies to the logic below, where we also check the chainId before checking `isFormItemAllowed`